pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -190,8 +197,29 @@ func (c *Controller) removeStaleGatewayRoutes() error {
 		desiredPodCIDRs = append(desiredPodCIDRs, podCIDRs...)
 	}
 
+	desiredClusterIPSvcIPs := map[string]bool{}
+	if c.proxyAll && runtime.IsWindowsPlatform() {
```

Comment:
Linux has same requirement, I think it does this in routeClient or haven't handled it yet. Anyway it's not proper to calculate service IPs in node_route_controller.go, I think we should handle it in some other common place for both Linux and Windows. Maybe remove it here and I could think about it for Linux and then Windows can share the cleanup logic.

Reply:
I added a TODO comment here. I will rebase and remove it when such PR is merged. Otherwise I think function is not complete or working.